MediaPushPlugin/src/main/java/io/antmedia/plugin/MediaPushPlugin.java

```diff
@@ -107,7 +109,31 @@ public Result startMediaPush(String streamId, String websocketUrl, Endpoint requ
 		customModification(driver);
 		JavascriptExecutor js = (JavascriptExecutor) driver;
 		js.executeScript(String.format("window.postMessage({ command:  'WR_START_BROADCASTING', streamId: '%s', websocketURL: '%s', width: '%s', height: '%s' }, '*')", streamId, websocketUrl, request.getWidth(), request.getHeight()));
-		return new Result(true, streamId, "Media Push started");
+
+		// wait until stream is started
+		timeout = 20;
+		while (true) {
+			LogEntries logEntries = driver.manage().logs().get("browser");
+			for (LogEntry entry : logEntries) {
+				if (entry.getMessage().contains("publish_started")) {
```

Comment:
I recommend you to not use trues for the main functionality. Please provide permanent and robust solutions.

Reply:
When I checked selenium documentation I couldn't find 2 way communication. Let me check again for a more reliable way.